flow-server/src/main/resources/vite.generated.ts

```diff
@@ -62,49 +104,25 @@ function buildSWPlugin(): PluginOption {
       config = resolvedConfig;
     },
     async buildStart() {
-      const includedPluginNames = [
-        'alias',
-        'vite:resolve',
-        'vite:esbuild',
-        'rollup-plugin-dynamic-import-variables',
-        'vite:esbuild-transpile',
-        'vite:terser'
-      ];
-      const rollupPlugins: rollup.Plugin[] = config.plugins.filter((p) => {
-        return includedPluginNames.includes(p.name);
-      });
-      rollupPlugins.push(
-        replace({
-          values: {
-            'process.env.NODE_ENV': JSON.stringify(config.mode),
-            ...config.define
-          },
-          preventAssignment: true
-        })
-      );
-
-      const rollupOutput: rollup.OutputOptions = {
-        file: path.resolve(frontendBundleFolder, 'sw.js'),
-        format: 'es',
-        exports: 'none',
-        sourcemap: config.command === 'serve' || config.build.sourcemap,
-        inlineDynamicImports: true
-      };
-
-      const rollupConfig: rollup.RollupOptions = {
-        input: path.resolve(settings.clientServiceWorkerSource),
-        output: rollupOutput,
-        plugins: rollupPlugins
-      };
-
-      const bundle = await rollup.rollup(rollupConfig);
-      try {
-        await bundle.write(rollupOutput);
-      } finally {
-        await bundle.close();
+      if (devMode) {
+        const { output } = await build('generate');
+        swObj.code = output[0].code;
+        swObj.map =  output[0].map;
+      } else {
+        await build('write');
       }
-    }
-  };
+    },
+    async load(id) {
+      if (id.endsWith('sw.js')) {
+        return '';
+      }
+    },
+    async transform(_code, id) {
+      if (id.endsWith('sw.js')) {
+        return swObj;
```

Comment:
Vite caches transform results so the bundle can be confidently generated here:

```suggestion
        const { output } = await build('generate');
        return output[0]; 
```

This way the bundle generation will be delayed until the first `sw.js` request and furthermore no need to maintain `swObj`.

Reply:
Ah, didn't notice the comment that generating `sw.js` in `transform` actually causes an error.

Reply:
Calling `build('generate')` in the `transform` hook give me the same error mentioned here
https://github.com/vaadin/flow/issues/13115#issuecomment-1047692328

That's the reason why i moved it in a different hook

Reply:
> Ah, sorry, didn't notice the comment that generating `sw.js` in `transform` actually causes an error.

No problem, the comment was somehow "hidden" by bot comments :)